Observer/TescoPaymentObserver.php

```diff
@@ -0,0 +1,99 @@
+<?php
+namespace Omise\Payment\Observer;
+
+use Magento\Framework\Event\ObserverInterface;
+use Magento\Payment\Model\InfoInterface;
+use Magento\Framework\Event\Observer;
+
+class TescoPaymentObserver implements ObserverInterface
+{
+    /**
+    * @var \Magento\Checkout\Model\Session
+    */
+    private $_checkoutSession;
+
+    /**
+    * @var \Magento\Framework\App\Config\ScopeConfigInterface
+    */
+    private $_scopeConfig;    
+    
+    /**
+    * @var  \Magento\Framework\Mail\Template\TransportBuilder
+    */
+    private $_transportBuilder;
+    
+    /**
+    * @var  \Omise\Payment\Helper
+    */
+    private $_helper;
+
+    public function __construct(
+        \Omise\Payment\Helper\OmiseHelper $helper,
+        \Magento\Checkout\Model\Session $checkoutSession,
+        \Magento\Framework\Mail\Template\TransportBuilder $transportBuilder,
+        \Magento\Framework\App\Config\ScopeConfigInterface $scopeConfig
+    ) {
+        $this->_scopeConfig = $scopeConfig;
```

Comment:
@jacstn just minor thing, but might be a bit easier to read with
```php
public function __construct(
    \Omise\Payment\Helper\OmiseHelper $helper,
    \Magento\Checkout\Model\Session $checkoutSession,
    \Magento\Framework\Mail\Template\TransportBuilder $transportBuilder,
    \Magento\Framework\App\Config\ScopeConfigInterface $scopeConfig
) {
    $this->_helper           = $helper;
    $this->_checkoutSession  = $checkoutSession;
    $this->_transportBuilder = $transportBuilder;
    $this->_scopeConfig      = $scopeConfig;
}
```
(just reorder of those properties assigning  to be in the same order as method's argument)